decentraland/unity-renderer
# fix: Fix CJK IME for WebGL

## What does this PR change?
The WebGL build was not showing the [IME](https://docs.unity3d.com/Manual/IMEInput.html) panel (it worked fine only for Unity editor and Desktop builds).

In order to solve this, we found this implementation by a third person on github that solves exactly our issue: https://github.com/kou-yeung/WebGLInput

So we have made a **[FORK](https://github.com/decentraland/webgl-ime-input)** of this repo and have added it into the project through the Package Manager, assigned the `WebGLInput` component to the input of our `ChatHUD` prefab and it works! We have available the IME panel also from the browser.

## Code review notes
All the code and assets included in this PR comes from the Unity package implemented by another person external to Decentraland in this external repo: https://github.com/kou-yeung/WebGLInput.

## How to test the changes?
1. Set your keyboard configuration as chineese, japaneese or korean (maybe you have to install these languages in your computer):

![image](https://user-images.githubusercontent.com/64659061/189394417-b31b489f-47ab-4006-80ac-2945d8fb72e2.png)

2. Go to: https://play.decentraland.zone/?renderer-branch=fix/fix-CJK-IME-tool-for-WebGL
3. Open the chat.
4. Try to write something.
5. Notice the IME panel is showed and let you select the different asian symbols.

![image](https://user-images.githubusercontent.com/64659061/189394542-02bc8fa1-52dc-4c5b-9b6b-c0019d8df40c.png)

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
